### PR TITLE
HOSTEDCP-1609: [ARO-HCP] Support Azure diagnostics

### PR DESCRIPTION
**Pull request by fxierh**
Managed diagnostics:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55434/rehearse-55434-periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7/1823796317374124032

No diagnostics settings:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55434/rehearse-55434-periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-cilium-guest-f7/1823796317516730368

UserManaged diagnostics:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55434/rehearse-55434-periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7/1823796317449621504

### CODE REVIEW

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-cilium-guest-f7

**Comment by fxierh** [comment]
/cc @jianlinliu 

Could you please take a look ?

File: ci-operator/step-registry/azure/provision/storage-account/azure-provision-storage-account-commands.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+RESOURCE_NAME_PREFIX="$(echo -n "$PROW_JOB_ID" | sha256sum | cut -c -15)"
+
+az --version
+az cloud set --name AzureCloud
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+set -x
+
+echo "Creating NSG in its own RG"
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/provision/storage-account/azure-provision-storage-account-commands.sh:19]: Wrong message ?

Comment (fxierh) [reply ci-operator/step-registry/azure/provision/storage-account/azure-provision-storage-account-commands.sh]: Updated. Thanks.

**Comment by jianlinliu** [comment]
cc @jinyunma to review here.

**Pull request merged**